spec/javascript/components/applications/application_form/PaperApplicationForm.test.js

```diff
@@ -0,0 +1,41 @@
+/* global wait */
+import React from 'react'
+import { clone } from 'lodash'
+import { mount } from 'enzyme'
+
+import PaperApplicationForm from 'components/applications/application_form/PaperApplicationForm'
+import listing from '../../../fixtures/listing'
+import application from '../../../fixtures/domain_application'
+
+const mockSubmitApplication = jest.fn()
+
+jest.mock('apiService', () => {
+  return {
+    submitApplication: async (data) => {
+      mockSubmitApplication(data)
+      return true
+    }
+  }
+})
+
+describe('PaperApplicationForm', () => {
+  describe('should validate fields correctly: ', () => {
+    test('Annual Income', async () => {
+      const applicationWithInvalidAnnualIncome = clone(application)
+      applicationWithInvalidAnnualIncome['annual_income'] = 'foo'
+
+      const wrapper = mount(
+        <PaperApplicationForm
+          listing={listing}
+          application={applicationWithInvalidAnnualIncome}
+        />
+      )
+
+      wrapper.find('form').first().simulate('submit')
+
+      await wait(100)
+
+      expect(wrapper).toMatchSnapshot()
```

Comment:
@callaghanc and @chrisdolendo Unrelated to this PR, but a question I have in general: Would it be possible to do the same sort of checks that we do in e2e tests for the presence of e.g. a validation error in these snapshot tests?

For example, in the ApplicationNewPage "should fail if required fields are missing" test, we have this expectation. 
```
const errors = await page.$$eval('.form-group.error span.error', divs => divs.map(d => d.textContent))
expect(errors).toContain('Please enter a First Name')
```

we could add something like that in addition to checking that it matches the snapshot, to be a little more explicit about what we're expecting.

Reply:
already chatted about this in slack, but check out the new test updates i added that do something very similar to this